omnigibson/object_states/overlaid.py

```diff
@@ -20,13 +24,57 @@
 # Subsample cloth particle points to fit a convex hull for efficiency purpose
 m.N_POINTS_CONVEX_HULL = 1000
 
+# z-offset for sampling
+m.SAMPLING_Z_OFFSET = 0.01
+
 class Overlaid(KinematicsMixin, RelativeObjectState, BooleanState):
     @staticmethod
     def get_dependencies():
         return KinematicsMixin.get_dependencies() + RelativeObjectState.get_dependencies() + [Touching]
 
-    def _set_value(self, other, new_value, use_ray_casting_method=False):
-        raise NotImplementedError("Overlaid state currently does not support setting.")
+    def _set_value(self, other, new_value):
+        if not new_value:
+            raise NotImplementedError("Overlaid does not support set_value(False)")
+
+        if not (self.obj.prim_type == PrimType.CLOTH and other.prim_type == PrimType.RIGID):
+            raise ValueError("Overlaid state requires obj1 is cloth and obj2 is rigid.")
+
+        sampling_success = False
+        state = self._simulator.dump_state(serialized=False)
+
+        # Get the top center of the object below
+        aabb_low, aabb_hi = other.states[AABB].get_value()
+        top_center = (aabb_low + aabb_hi) / 2 + np.array([0., 0., (aabb_hi - aabb_low)[2] / 2.0])
```

Comment:
super nit, couldn't we just use aabb_hi[2] as the z value?

Reply:
done